scylla-cql/src/frame/response/result.rs

```diff
@@ -441,13 +442,19 @@ pub enum Result {
     SchemaChange(SchemaChange),
 }
 
-fn deser_table_spec(buf: &mut &[u8]) -> StdResult<TableSpec, ParseError> {
-    let ks_name = types::read_string(buf)?.to_owned();
-    let table_name = types::read_string(buf)?.to_owned();
-    Ok(TableSpec {
-        ks_name,
-        table_name,
-    })
+// Only allocates a new `TableSpec` if one is not yet given.
+// TODO: consider equality check between known and deserialized spec.
+fn deser_table_spec(
+    buf: &mut &[u8],
+    known_spec: Option<TableSpec>,
+) -> StdResult<TableSpec, ParseError> {
+    let ks_name = types::read_string(buf)?;
+    let table_name = types::read_string(buf)?;
+
+    Ok(known_spec.unwrap_or_else(|| TableSpec {
+        ks_name: ks_name.to_owned(),
+        table_name: table_name.to_owned(),
+    }))
```

Comment:
I'd like those equality checks to be a part of this PR so that we have more chances of catching any issues caused by this change.

As you know CQL protocol allows table spec to be either global or per column.
Do you know why that is? I assume there is some reason for the server to not always send the global one, and right now this PR assumes there is no reason and we can just use table spec of last column everywhere.

If there really is no reason for those per-column specs to exist, then that should be explained in the comment, and there should still be panicking checks in case this assumption turns out to be false.

Reply:
Right, let's mind the Chesterton’s Fence: _don’t ever take down a fence until you know why it was put up._ 
I'll investigate those per-column table specs.

Reply:
[Python driver makes the same assumption](https://github.com/scylladb/python-driver/blob/f3567166a2c151a1d8a44e1281b17bd612b2032d/cassandra/query.py#L557-L560), i.e. uses the table spec of the first column for all columns.

Reply:
This may be a good question for #engineering channel on Slack. I'd like to know when will Scylla / Cassandra send global spec and when per-column spec, and if it's posibble for columns specs to differ. If it's not possible, then learning a bit of history would be beneficial here imo: was it possible in the past? What was the case for it?

Reply:
@nyh This is what I've talked to you about.

Reply:
One more thing to consider: how likely is it that in some future version Cassandra / Scylla adds some form of joins / multi table queries?
AFAIK Cassandra already added ACID transactions (using their ACCORD algorithm), it doesn't seem so improbable for them to add something that queries more than 1 table in the future.
As those structs you modify are public, supporting this would require a breaking change. 

Do you think we could use Cow / Arc or something else to make this future-proof? That way we could have global table spec in `ResultMetadata`, but also have per-column spec if necessary.

Reply:
Adding joins to CQL would be such a giant change in ScyllaDB - wide-column DBs aren't made for joins - that I strongly doubt it will ever happen.
`Arc`s and `Cow`s have considerable overhead that I don't deem worth incurring here just due to an extremely unprobable scenario.